File: test/domain/shelter_link_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:shelter_app/features/shared/domain/shelter_link.dart';

void main() {
  group('buildShelterUrl', () {
    test('pads a short number to 4 digits', () {
      expect(
        buildShelterUrl('573/26'),
        'https://napaluchu.waw.pl/animal/0573-26p/',
      );
    });

    test('leaves an already 4-digit number as-is', () {
      expect(
        buildShelterUrl('2222/26'),
        'https://napaluchu.waw.pl/animal/2222-26p/',
      );
    });

    test('trims surrounding whitespace', () {
      expect(
        buildShelterUrl(' 573/26 '),
        'https://napaluchu.waw.pl/animal/0573-26p/',
      );
    });

    test('returns null for null or empty', () {
      expect(buildShelterUrl(null), isNull);
      expect(buildShelterUrl(''), isNull);
      expect(buildShelterUrl('   '), isNull);
    });

    test('returns null when shelterId does not match the number/year shape', () {
      expect(buildShelterUrl('S010'), isNull);
      expect(buildShelterUrl('573-26'), isNull);
      expect(buildShelterUrl('573/26/1'), isNull);
    });
  });
}

File: api/__tests__/shelter-link.test.ts
import { buildShelterUrl } from '../_lib/shelter-link';

describe('buildShelterUrl', () => {
	it('pads a short number to 4 digits', () => {
		expect(buildShelterUrl('573/26')).toBe('https://napaluchu.waw.pl/animal/0573-26p/');
	});

	it('leaves an already 4-digit number as-is', () => {
		expect(buildShelterUrl('2222/26')).toBe('https://napaluchu.waw.pl/animal/2222-26p/');
	});

	it('trims surrounding whitespace', () => {
		expect(buildShelterUrl(' 573/26 ')).toBe('https://napaluchu.waw.pl/animal/0573-26p/');
	});

	it('returns null for null, undefined, or empty', () => {
		expect(buildShelterUrl(null)).toBeNull();
		expect(buildShelterUrl(undefined)).toBeNull();
		expect(buildShelterUrl('')).toBeNull();
	});

	it('returns null when shelterid does not match the number/year shape', () => {
		expect(buildShelterUrl('S010')).toBeNull();
		expect(buildShelterUrl('573-26')).toBeNull();
		expect(buildShelterUrl('573/26/1')).toBeNull();
	});
});
